x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/patch_rules_bulk_route.ts

```diff
@@ -106,14 +118,20 @@ export const patchRulesBulkRoute = (router: IRouter, ml: SetupPlugins['ml']) =>
               throwHttpError(await mlAuthz.validateRuleType(type));
             }
 
-            const existingRule = await readRules({ alertsClient, ruleId, id });
-            if (existingRule?.params.type) {
+            const foundRule = foundRules?.find((rule) => rule.params.ruleId === ruleId);
+            let existingRule = foundRule != null ? [foundRule] : null;
```

Comment:
You could also combine these statements to get rid of the `let` ...

Reply:
We do need the check there since the type would be `RuleAlertType | RuleAlertType[] | null` which would cause the below checks to increase in complexity since we'd have to check if it's an array first before attempting to access etc.. But thanks @madirey for the suggestion! I will combine the two.